napari/viewer.py

```diff
@@ -45,3 +45,8 @@ def __init__(self, title='napari', ndisplay=2, order=None):
         self.window = Window(qt_viewer)
         self.screenshot = self.window.qt_viewer.screenshot
         self.update_console = self.window.qt_viewer.console.push
+
+    def update(self, func, *args, **kwargs):
+        t = QtUpdateUI(func, *args, **kwargs)
+        self.window.qt_viewer.pool.start(t)
+        return self.window.qt_viewer.pool  # returns threadpool object
```

Comment:
don't have an opinion, just curious: why did you choose to return the full threadpool instead of the new thread?

Reply:
We don't need to return threadpool. With current implementation threadpool is the one who creates and run thread from our runnable. I do not know an intuitive way to access created thread object. This pattern is good for run and forget type of threads. They are _like_ daemon threads, will not be affected by termination of the origin thread. More meaningful methods are provided on threadpool object, hence i returned that. Like there are methods to get information about pool and prevent execution of queued runnables. I am open to suggestions, wanna return something else? @tlambert03

Reply:
no no, it makes sense to me.  Really was just curious